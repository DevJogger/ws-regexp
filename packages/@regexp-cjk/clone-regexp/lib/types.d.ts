import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';
export interface ICloneRegexp<T extends RegExp = RegExp> {
    (inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
}
export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags {
}
export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore {
    /**
     * allow change cloneRegexp function
     */
    cloneRegexp?: ICloneRegexp<T>;
    disableDetectRegexpClone?: boolean;
}
export declare type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
    clone?(...argv: any[]): any;
} ? T : T & {
    clone?(options?: ICloneRegexpOptionsCore, ...argv: any[]): T;
};
