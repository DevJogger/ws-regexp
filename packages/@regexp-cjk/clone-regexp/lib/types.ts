import { Options } from 'clone-regexp';
import { IOptionsRewriteFlags } from '@regexp-cjk/rewrite-flags';

export interface ICloneRegexp<T extends RegExp = RegExp>
{
	(inputRegExp: T | RegExp, options?: ICloneRegexpOptionsCore, ...argv): T
}

export interface ICloneRegexpOptionsCore extends Options, IOptionsRewriteFlags
{

}

export interface ICloneRegexpOptionsCustom<T extends RegExp = RegExp>
{
	/**
	 * allow change cloneRegexp function
	 */
	cloneRegexp?: ICloneRegexp<T>,

	disableDetectRegexpClone?: boolean,

	resetLastIndex?: boolean,

	lastIndex?: number,
}

export interface ICloneRegexpOptions<T extends RegExp = RegExp> extends ICloneRegexpOptionsCore, ICloneRegexpOptionsCustom<T>
{

}

export type IRegExpWithClone<T extends RegExp = RegExp> = T extends {
	clone?(...argv)
} ? T : T & {
	clone?(options?: ICloneRegexpOptionsCore, ...argv): T
}
