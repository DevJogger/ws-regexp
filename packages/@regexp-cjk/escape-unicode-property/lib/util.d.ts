import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
