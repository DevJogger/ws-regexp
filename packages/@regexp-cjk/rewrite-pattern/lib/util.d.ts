/**
 * Created by user on 2020/6/8.
 */
import { IOptionsRewritePattern } from '../index';
export declare function rewritePatternOptions(options?: IOptionsRewritePattern, flags?: string): IOptionsRewritePattern;
export declare function handleOptions(options: IOptionsRewritePattern, flags?: string): {
    options: IOptionsRewritePattern;
    flags: string;
};
