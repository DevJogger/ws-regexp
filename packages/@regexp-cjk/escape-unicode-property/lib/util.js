"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const util_1 = require("@regexp-cjk/rewrite-pattern/lib/util");
const lodash_1 = require("lodash");
function handleOptions(options, flags) {
    return (0, util_1.handleOptions)((0, lodash_1.merge)({}, options, {
        unicodePropertyEscape: true,
        rewriteFlags: {
            unicode: true,
        },
    }), flags !== null && flags !== void 0 ? flags : 'u');
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map