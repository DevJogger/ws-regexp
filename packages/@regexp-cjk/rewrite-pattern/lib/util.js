"use strict";
/**
 * Created by user on 2020/6/8.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = exports.rewritePatternOptions = void 0;
const lodash_1 = require("lodash");
const rewrite_flags_1 = require("@regexp-cjk/rewrite-flags");
function rewritePatternOptions(options, flags) {
    var _a;
    return (0, lodash_1.defaults)({}, options, {
        dotAllFlag: true,
        lookbehind: true,
        namedGroup: true,
        useUnicodeFlag: (_a = flags === null || flags === void 0 ? void 0 : flags.includes) === null || _a === void 0 ? void 0 : _a.call(flags, 'u'),
    });
}
exports.rewritePatternOptions = rewritePatternOptions;
function handleOptions(options, flags) {
    var _a;
    flags = (_a = options === null || options === void 0 ? void 0 : options.flags) !== null && _a !== void 0 ? _a : flags;
    if (options === null || options === void 0 ? void 0 : options.rewriteFlags) {
        flags = (0, rewrite_flags_1.rewriteFlags)(flags, options.rewriteFlags);
    }
    options = rewritePatternOptions(options, flags);
    delete options.flags;
    return {
        options,
        flags,
    };
}
exports.handleOptions = handleOptions;
//# sourceMappingURL=util.js.map