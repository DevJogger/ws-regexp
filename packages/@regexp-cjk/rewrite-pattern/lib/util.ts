/**
 * Created by user on 2020/6/8.
 */

import { defaults } from 'lodash';
import { IOptionsRewritePattern } from '../index';
import { rewriteFlags } from '../flags';

export function rewritePatternOptions(options?: IOptionsRewritePattern, flags?: string): IOptionsRewritePattern
{
	return defaults<IOptionsRewritePattern, IOptionsRewritePattern, IOptionsRewritePattern>({}, options, {
		dotAllFlag: true,
		lookbehind: true,
		namedGroup: true,
		useUnicodeFlag: flags?.includes?.('u'),
	})
}

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	flags = options?.flags ?? flags;

	if (options?.rewriteFlags)
	{
		flags = rewriteFlags(flags, options.rewriteFlags);
	}

	options = rewritePatternOptions(options, flags)

	return {
		options,
		flags,
	}
}
