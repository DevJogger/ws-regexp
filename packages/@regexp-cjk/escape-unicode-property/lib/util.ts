import { IOptionsRewritePattern } from '@regexp-cjk/rewrite-pattern';
import { handleOptions as _handleOptions } from '@regexp-cjk/rewrite-pattern/lib/util';
import { merge } from 'lodash';

export function handleOptions(options: IOptionsRewritePattern, flags?: string)
{
	return _handleOptions(merge({}, options, {
		unicodePropertyEscape: true,
		rewriteFlags: {
			unicode: true,
		},
	}), flags ?? 'u');
}
