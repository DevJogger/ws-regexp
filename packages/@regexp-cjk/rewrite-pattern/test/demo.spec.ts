import { inspect } from "util";
import rewritePattern, { rewritePatternByRegExp, rewriteRegExp, IOptionsRewritePattern } from '../index';
import { handleOptions } from '../lib/util';

interface ITestCase
{
	args: [string, string?, IOptionsRewritePattern?],
	expected?: string,
	error?: boolean,
}

describe(`simple`, () =>
{
	([
		{
			args: ['foo.bar'],
		},
		{
			args: ['foo.bar', 'u'],
		},
		{
			args: ['[\\u{1D306}-\\u{1D308}a-z]', 'u'],
		},
		{
			args: ['[\\u{1D306}-\\u{1D308}a-z]', 'ui'],
		},
		{
			args: ['.', 'u'],
		},
		{
			args: ['.', 'u', {
				'dotAllFlag': false,
			}],
		},
		{
			args: ['.', '', {
				'dotAllFlag': true,
			}],
		},
		{
			args: ['.', 'su', {
				'dotAllFlag': false,
			}],
		},
		{
			args: ['.', 'su', {
				'dotAllFlag': true
			}],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u'],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': true
			}],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': false
			}],
		},
		{
			args: ['(?<=.)a'],
		},
		{
			args: ['(?<=.)a', '', {
				'lookbehind': true
			}],
		},
		{
			args: ['(?<=.)a', '', {
				'lookbehind': false
			}],
			error: true,
		},
		{
			args: ['(?<name>.)\\k<name>'],
			//expected: `(.)\\1`,
		},
		{
			args: ['(?<name>.)\\k<name>', '', {
				'namedGroup': true
			}],
			//expected: `(.)\\1`,
		},
		{
			args: ['(?<name>.)\\k<name>', '', {
				'namedGroup': false
			}],
			error: true,
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': true,
				'useUnicodeFlag': true
			}],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': true,
				'useUnicodeFlag': false
			}],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': false,
				'useUnicodeFlag': true
			}],
		},
		{
			args: ['\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u', {
				'unicodePropertyEscape': false,
				'useUnicodeFlag': false
			}],
		},
		{
			args: [],
		},
		{
			args: [],
		},
		{
			args: [],
		},
		{
			args: [],
		},
		{
			args: [],
		},
	] as ITestCase[]).forEach(testCase =>
	{
		const { args, expected } = testCase;

		if (!args.length)
		{
			return;
		}

		describe((testCase.error ? 'ThrowError ' : '') + inspect(args), () =>
		{
			if (args[2])
			{
				test(`options`, () =>
				{
					let actual = handleOptions(args[2], args[1]);
					expect(actual).toMatchSnapshot();
				})
			}

			if (testCase.error)
			{
				test(`throw error`, () =>
				{
					expect(() => rewritePattern(...args)).toThrowError();

					let re = new RegExp(args[0], args[1])

					expect(() => rewritePatternByRegExp(re, args[2])).toThrowError();

					expect(() => rewriteRegExp(re, args[2])).toThrowError();
				})

				return;
			}

			test(`test`, () =>
			{

				let actual = rewritePattern(...args);

				let re = new RegExp(args[0], args[1])

				let actual2 = rewritePatternByRegExp(re, args[2]);

				let actual3 = rewriteRegExp(re, args[2]);

				console.log(actual)
				console.log(actual3)

				if (typeof expected !== 'undefined')
				{
					expect(actual).toStrictEqual(expected);
				}

				expect(actual).toStrictEqual(actual2);
				expect(actual).toStrictEqual(actual3.source);

				//expect(actual).toBeInstanceOf(Date);

				expect(actual).toMatchSnapshot();
				expect(actual3).toMatchSnapshot();

			});
		})



	})
	;

})
