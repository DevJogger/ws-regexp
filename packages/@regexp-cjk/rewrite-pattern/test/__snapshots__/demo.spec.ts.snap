// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": false,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false }
] test 1`] = `"\\p{Script_Extensions=Anatolian_Hieroglyphs}"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false }
] test 2`] = `/\\\\p\\{Script_Extensions=Anatolian_Hieroglyphs\\}/u`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: false }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": false,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: false }
] test 1`] = `"\\p{Script_Extensions=Anatolian_Hieroglyphs}"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: false }
] test 2`] = `/\\\\p\\{Script_Extensions=Anatolian_Hieroglyphs\\}/u`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: true }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": false,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: true }
] test 1`] = `"\\p{Script_Extensions=Anatolian_Hieroglyphs}"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: false, useUnicodeFlag: true }
] test 2`] = `/\\\\p\\{Script_Extensions=Anatolian_Hieroglyphs\\}/u`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true }
] test 1`] = `"[\\u{14400}-\\u{14646}]"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true }
] test 2`] = `/\\[\\\\u\\{14400\\}-\\\\u\\{14646\\}\\]/u`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: false }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: false }
] test 1`] = `"(?:\\uD811[\\uDC00-\\uDE46])"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: false }
] test 2`] = `/\\(\\?:\\\\uD811\\[\\\\uDC00-\\\\uDE46\\]\\)/u`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: true }
] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "unicodePropertyEscape": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: true }
] test 1`] = `"[\\u{14400}-\\u{14646}]"`;

exports[`simple [
  '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}',
  'u',
  { unicodePropertyEscape: true, useUnicodeFlag: true }
] test 2`] = `/\\[\\\\u\\{14400\\}-\\\\u\\{14646\\}\\]/u`;

exports[`simple [ '(?<=.)a' ] test 1`] = `"(?<=.)a"`;

exports[`simple [ '(?<=.)a' ] test 2`] = `/\\(\\?<=\\.\\)a/`;

exports[`simple [ '(?<=.)a', '', { lookbehind: true } ] options 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple [ '(?<=.)a', '', { lookbehind: true } ] test 1`] = `"(?<=.)a"`;

exports[`simple [ '(?<=.)a', '', { lookbehind: true } ] test 2`] = `/\\(\\?<=\\.\\)a/`;

exports[`simple [ '(?<name>.)\\\\k<name>' ] test 1`] = `"(.)\\1"`;

exports[`simple [ '(?<name>.)\\\\k<name>' ] test 2`] = `/\\(\\.\\)\\\\1/`;

exports[`simple [ '(?<name>.)\\\\k<name>', '', { namedGroup: true } ] options 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple [ '(?<name>.)\\\\k<name>', '', { namedGroup: true } ] test 1`] = `"(.)\\1"`;

exports[`simple [ '(?<name>.)\\\\k<name>', '', { namedGroup: true } ] test 2`] = `/\\(\\.\\)\\\\1/`;

exports[`simple [ '.', '', { dotAllFlag: true } ] options 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple [ '.', '', { dotAllFlag: true } ] test 1`] = `"."`;

exports[`simple [ '.', '', { dotAllFlag: true } ] test 2`] = `/\\./`;

exports[`simple [ '.', 'su', { dotAllFlag: false } ] options 1`] = `
{
  "flags": "su",
  "options": {
    "dotAllFlag": false,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [ '.', 'su', { dotAllFlag: false } ] test 1`] = `"[\\0-\\t\\x0B\\f\\x0E-\\u2027\\u202A-\\u{10FFFF}]"`;

exports[`simple [ '.', 'su', { dotAllFlag: false } ] test 2`] = `/\\[\\\\0-\\\\t\\\\x0B\\\\f\\\\x0E-\\\\u2027\\\\u202A-\\\\u\\{10FFFF\\}\\]/su`;

exports[`simple [ '.', 'su', { dotAllFlag: true } ] options 1`] = `
{
  "flags": "su",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [ '.', 'su', { dotAllFlag: true } ] test 1`] = `"[\\0-\\u{10FFFF}]"`;

exports[`simple [ '.', 'su', { dotAllFlag: true } ] test 2`] = `/\\[\\\\0-\\\\u\\{10FFFF\\}\\]/su`;

exports[`simple [ '.', 'u' ] test 1`] = `"[\\0-\\t\\x0B\\f\\x0E-\\u2027\\u202A-\\u{10FFFF}]"`;

exports[`simple [ '.', 'u' ] test 2`] = `/\\[\\\\0-\\\\t\\\\x0B\\\\f\\\\x0E-\\\\u2027\\\\u202A-\\\\u\\{10FFFF\\}\\]/u`;

exports[`simple [ '.', 'u', { dotAllFlag: false } ] options 1`] = `
{
  "flags": "u",
  "options": {
    "dotAllFlag": false,
    "lookbehind": true,
    "namedGroup": true,
    "useUnicodeFlag": true,
  },
}
`;

exports[`simple [ '.', 'u', { dotAllFlag: false } ] test 1`] = `"[\\0-\\t\\x0B\\f\\x0E-\\u2027\\u202A-\\u{10FFFF}]"`;

exports[`simple [ '.', 'u', { dotAllFlag: false } ] test 2`] = `/\\[\\\\0-\\\\t\\\\x0B\\\\f\\\\x0E-\\\\u2027\\\\u202A-\\\\u\\{10FFFF\\}\\]/u`;

exports[`simple [ '[\\\\u{1D306}-\\\\u{1D308}a-z]', 'u' ] test 1`] = `"[a-z\\u{1D306}-\\u{1D308}]"`;

exports[`simple [ '[\\\\u{1D306}-\\\\u{1D308}a-z]', 'u' ] test 2`] = `/\\[a-z\\\\u\\{1D306\\}-\\\\u\\{1D308\\}\\]/u`;

exports[`simple [ '[\\\\u{1D306}-\\\\u{1D308}a-z]', 'ui' ] test 1`] = `"[a-z\\u{1D306}-\\u{1D308}]"`;

exports[`simple [ '[\\\\u{1D306}-\\\\u{1D308}a-z]', 'ui' ] test 2`] = `/\\[a-z\\\\u\\{1D306\\}-\\\\u\\{1D308\\}\\]/iu`;

exports[`simple [ '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u' ] test 1`] = `"\\p{Script_Extensions=Anatolian_Hieroglyphs}"`;

exports[`simple [ '\\\\p{Script_Extensions=Anatolian_Hieroglyphs}', 'u' ] test 2`] = `/\\\\p\\{Script_Extensions=Anatolian_Hieroglyphs\\}/u`;

exports[`simple [ 'foo.bar' ] test 1`] = `"foo.bar"`;

exports[`simple [ 'foo.bar' ] test 2`] = `/foo\\.bar/`;

exports[`simple [ 'foo.bar', 'u' ] test 1`] = `"foo[\\0-\\t\\x0B\\f\\x0E-\\u2027\\u202A-\\u{10FFFF}]bar"`;

exports[`simple [ 'foo.bar', 'u' ] test 2`] = `/foo\\[\\\\0-\\\\t\\\\x0B\\\\f\\\\x0E-\\\\u2027\\\\u202A-\\\\u\\{10FFFF\\}\\]bar/u`;

exports[`simple ThrowError [ '(?<=.)a', '', { lookbehind: false } ] options 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": false,
    "namedGroup": true,
    "useUnicodeFlag": false,
  },
}
`;

exports[`simple ThrowError [ '(?<name>.)\\\\k<name>', '', { namedGroup: false } ] options 1`] = `
{
  "flags": "",
  "options": {
    "dotAllFlag": true,
    "lookbehind": true,
    "namedGroup": false,
    "useUnicodeFlag": false,
  },
}
`;
